cost-analyzer/templates/query-service-deployment-template.yaml

```diff
@@ -0,0 +1,113 @@
+{{- if and (not .Values.agent) (not .Values.cloudAgent) (.Values.kubecostDeployment) }}
+{{- if and (.Values.kubecostDeployment.queryServiceReplicas) (.Values.kubecostDeployment.queryServiceReplicas.enabled) }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "query-service.fullname" . }}
+  namespace: {{ .Release.Name }}
+  labels:
+    {{ include "query-service.commonLabels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.kubecostDeployment.queryServiceReplicas.replicas | default 1 }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: query-service
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app: query-service
+  serviceName: {{ .Release.Name }}-query-service-load-balancer
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: query-service
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app: query-service
+    spec:
+      restartPolicy: Always
+      serviceAccountName: {{ template "query-service.serviceAccountName" . }}
+      volumes:
+        {{- $etlBackupBucketSecret := "" }}
+        {{- if .Values.kubecostModel.etlBucketConfigSecret }}
+            {{- $etlBackupBucketSecret = .Values.kubecostModel.etlBucketConfigSecret }}
+        {{- else if and .Values.global.thanos.enabled (ne (typeOf .Values.kubecostModel.etlBucketConfigSecret) "string") }}
+            {{- $etlBackupBucketSecret = .Values.thanos.storeSecretName }}
+        {{- end }}
+        {{- if $etlBackupBucketSecret }}
+        - name: etl-bucket-config
+          secret:
+           defaultMode: 420
+           secretName: {{ $etlBackupBucketSecret }}
+        {{- end }}
+      initContainers:
+        - name: config-db-perms-fix
+          image: busybox
+          command: ["sh", "-c", "/bin/chmod -R 777 /var/configs && /bin/chmod -R 777 /var/db"]
+          volumeMounts:
+            - name: persistent-configs
+              mountPath: /var/configs
+            - name: persistent-db
+              mountPath: /var/db
+          securityContext:
+            runAsUser: 0
+      containers:
+        - name: query-service
+          image: {{ .Values.kubecostModel.fullImageName }}
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 9003
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            failureThreshold: 200
+          imagePullPolicy: Always
+          args: ["query-service"]
+          ports:
+            - name: tcp-model
+              containerPort: 9003
+              protocol: TCP
+            - name: tcp-frontend
+              containerPort: 9090
+              protocol: TCP
```

Comment:
Does query service actually have a frontend (or anything) running on 9090?

Reply:
Negative, I'll take that out.